src/python/pants/testutil/goal_rule_test_base.py

```diff
@@ -38,6 +38,7 @@ def setUp(self):
     def execute_rule(
         self,
         args: Optional[Iterable[str]] = None,
+        global_args: Optional[Iterable[str]] = None,
```

Comment:
Why make this change? Oh, so that you use the option shorthand? Check all the other files in this class, those might need updating too. 

I'm not certain we want to make this change because it's not that much extra typing to do `--repl-shell` than `--shell` and it's not very clear (to me) what the difference between args vs. global args are. I had to look at the implementation to figure it out that they're all args, only separated by the `goal_cls.name`.

Reply:
This wasn't added to make `--repl-shell` work, it was added because the `--backend-packages2` arg that was already in the test broke now that `ReplOptions` has an option defined on it.

Reply:
Okay, sounds reasonable. What do you think about changing `args` to be `goal_args`? So we have `global_args` and `goal_args`. (Genuine question, I'm not convinced that's a win.)

Reply:
I think that's clearer, but changing the name would also involve a bunch of unrelated changes everywhere we assume the use of a kwarg called `args`, so I'd rather not do it as part of this commit.